code/game/gamemodes/miniantags/demons/shadow_demon/shadow_demon.dm

```diff
@@ -57,6 +59,34 @@
 	max_integrity = 100
 	light_color = "#AAD84B"
 	anchored = TRUE
+	/// Amount of SSobj ticks (Roughly 2 seconds) since the last hallucination proc'd
+	var/time_since_last_hallucination = 0
+
+/obj/structure/shadowcocoon/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/structure/shadowcocoon/process()
+	time_since_last_hallucination++
+	for(var/atom/to_darken in range(4, src))
+		if(prob(60) || !length(to_darken.light_sources))
+			continue
+		if(iswelder(to_darken) && length(to_darken.light_sources))
+			var/obj/item/weldingtool/welder_to_darken = to_darken
+			welder_to_darken.remove_fuel(welder_to_darken.reagents.get_reagent_amount("fuel"))
```

Comment:
How does this interact with advanced welders?

Reply:
fine